plugins/common/wayfire/plugins/common/cairo-util.hpp

```diff
@@ -221,11 +221,18 @@ struct cairo_text_t
         return ret;
     }
 
+    cairo_text_t() = default;
+
     ~cairo_text_t()
     {
         cairo_free();
     }
 
+    cairo_text_t(const cairo_text_t &) = delete;
+    cairo_text_t(cairo_text_t &&) = default;
```

Comment:
I believe the move constructor and assignment should be deleted as well or implemented in a way that invalidates the members `cr` and `surface`, since they are raw pointers to object allocated by the Cairo C API (see the destructor below).

Reply:
(the same probably applied to `simple_texture_t` as well, where the destructor frees up the `tex` member)